Branch-SDK/src/io/branch/referral/SystemObserver.java

```diff
@@ -69,6 +79,42 @@ public String getURIScheme() {
 		return BLANK;
 	}
 	
+	@SuppressLint("NewApi")
+	public JSONArray getListOfApps() {
+		JSONArray arr = new JSONArray();
+		PackageManager pm = context_.getPackageManager();
+		
+		List<ApplicationInfo> packages = pm.getInstalledApplications(PackageManager.GET_META_DATA);
+		for (ApplicationInfo appInfo : packages) {
+			if ((appInfo.flags & ApplicationInfo.FLAG_SYSTEM) != 1) {
```

Comment:
Is this to filter out system apps? So the purpose of this new API is to retrieve all non-system apps installed on user's device, and see which ones have our SDK?